src/schemes/fluid/implicit_incompressible_sph/system.jl

```diff
@@ -415,32 +440,39 @@ function pressure_solve_iteration(system, u, u_ode, semi, time_step)
             sum_d_ij_pj[i, particle] += sum_dij_pj_[i]
         end
     end
+end
 
+function calculate_sum_term_values(system, u, u_ode, semi, time_step)
+    (; sum_term, pressure) = system
     # Calculate the large sum in eq. 13 of Ihmsen et al. (2013) for each particle (as `sum_term`)
     set_zero!(sum_term)
 
     foreach_system(semi) do neighbor_system
         calculate_sum_term!(sum_term, system, neighbor_system,
                             pressure, u, u_ode, semi, time_step)
     end
```

Comment:
```suggestion

        return system
    end
```
Note that a trailing `!` in the function name indicates not just that the first argument is modified, but also that this argument is returned.
Please return the system in all of these functions with a trailing `!` in the name.